vmm/src/cpu.rs

```diff
@@ -919,12 +921,14 @@ impl CpuManager {
 
                                 #[cfg(feature = "kvm")]
                                 {
-                                    vcpu.lock().as_ref().unwrap().vcpu.set_immediate_exit(true);
-                                    if !matches!(vcpu.lock().unwrap().run(), Ok(VmExit::Ignore)) {
-                                        error!("Unexpected VM exit on \"immediate_exit\" run");
-                                        break;
+                                    if on_kvm {
```

Comment:
You can drop the first commit in the series and simplify this all by just doing `if matches!(self.hypervisor_type, HypervisorType::Kvm)`